csharp/src/Ice/Internal/OutgoingAsync.cs

```diff
@@ -620,6 +617,195 @@ protected override bool responseImpl(bool userThread, bool ok, bool invoke)
 
     public void runTimerTask() => cancel(new Ice.InvocationTimeoutException());
 
+    private int handleRetryAfterException(Exception ex)
```

Comment:
We should always qualify Exception in C#: either Ice.Exception or System.Exception.

Otherwise, you wonder if the Ice.Exception is accidental.

Reply:
Also fixed!